EasyReflectometry/sample/elements/layers/layer_area_per_molecule.py

```diff
@@ -87,43 +87,44 @@ def __init__(
         :param name: Name of the layer, defaults to "EasyLayerAreaPerMolecule"
         :param interface: Interface object, defaults to `None`
         """
-        scattering_length = neutron_scattering_length(molecular_formula)
+
+        # Create the solvated molecule and corresponding constraints
+        molecule = Material.from_pars(0.0, 0.0, name=molecular_formula, interface=interface)
+
         default_options = deepcopy(LAYER_AREA_PER_MOLECULE_DETAILS)
         del default_options['sl']['value']
         del default_options['isl']['value']
-        scattering_length_real = Parameter('scattering_length_real', scattering_length.real, **default_options['sl'])
-        scattering_length_imag = Parameter('scattering_length_imag', scattering_length.imag, **default_options['isl'])
-        sld_real = area_per_molecule_to_sld(scattering_length_real.raw_value, thickness.raw_value, area_per_molecule.raw_value)
-        sld_imag = area_per_molecule_to_sld(scattering_length_imag.raw_value, thickness.raw_value, area_per_molecule.raw_value)
-
-        material = Material.from_pars(sld_real, sld_imag, name=molecular_formula, interface=interface)
+        scattering_length_real = Parameter('scattering_length_real', 0.0, **default_options['sl'])
+        scattering_length_imag = Parameter('scattering_length_imag', 0.0, **default_options['isl'])
 
+        # Constrain the sld value for the molecule
         constraint = FunctionalConstraint(
-            dependent_obj=material.sld,
+            dependent_obj=molecule.sld,
             func=area_per_molecule_to_sld,
             independent_objs=[scattering_length_real, thickness, area_per_molecule],
         )
         thickness.user_constraints['area_per_molecule'] = constraint
         area_per_molecule.user_constraints['area_per_molecule'] = constraint
         scattering_length_real.user_constraints['area_per_molecule'] = constraint
 
+        # Constrain the isld value for the molecule
```

Comment:
I assume the "i" in isld stands for imaginary? I would write out imaginary then, or complex. Especially if we also write out what sld stands for :)

Reply:
added `real` and `imag` to variable names and in code comments